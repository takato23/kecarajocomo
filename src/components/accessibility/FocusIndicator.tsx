'use client';

import { useEffect, useState } from 'react';

export function FocusIndicator() {
  const [isKeyboardUser, setIsKeyboardUser] = useState(false);

  useEffect(() => {
    // Detect keyboard navigation
    const handleMouseDown = () => setIsKeyboardUser(false);
    const handleKeyDown = (e: KeyboardEvent) => {
      if (e.key === 'Tab') {
        setIsKeyboardUser(true);
      }
    };

    document.addEventListener('mousedown', handleMouseDown);
    document.addEventListener('keydown', handleKeyDown);

    // Add focus-visible class to body when keyboard navigation is detected
    if (isKeyboardUser) {
      document.body.classList.add('keyboard-navigation');
    } else {
      document.body.classList.remove('keyboard-navigation');
    }

    return () => {
      document.removeEventListener('mousedown', handleMouseDown);
      document.removeEventListener('keydown', handleKeyDown);
    };
  }, [isKeyboardUser]);

  return null;
}