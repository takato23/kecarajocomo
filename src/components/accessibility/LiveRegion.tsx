'use client';

import { useEffect, useRef } from 'react';

interface LiveRegionProps {
  message: string;
  politeness?: 'polite' | 'assertive';
  clearOnUnmount?: boolean;
}

export function LiveRegion({ 
  message, 
  politeness = 'polite',
  clearOnUnmount = true 
}: LiveRegionProps) {
  const regionRef = useRef<HTMLDivElement>(null);
  const previousMessage = useRef<string>('');

  useEffect(() => {
    // Only announce if message has changed
    if (message && message !== previousMessage.current && regionRef.current) {
      // Clear and re-set the content to ensure screen readers announce it
      regionRef.current.textContent = '';
      setTimeout(() => {
        if (regionRef.current) {
          regionRef.current.textContent = message;
        }
      }, 100);
      previousMessage.current = message;
    }
  }, [message]);

  useEffect(() => {
    return () => {
      if (clearOnUnmount && regionRef.current) {
        regionRef.current.textContent = '';
      }
    };
  }, [clearOnUnmount]);

  return (
    <div 
      ref={regionRef}
      role="status" 
      aria-live={politeness} 
      aria-atomic="true"
      className="sr-only"
    />
  );
}

// Hook for using live regions
export function useLiveRegion() {
  const regionRef = useRef<HTMLDivElement | null>(null);
  const mounted = useRef(false);

  useEffect(() => {
    if (!mounted.current) {
      // Create a live region element and append it to the body
      const liveRegion = document.createElement('div');
      liveRegion.setAttribute('role', 'status');
      liveRegion.setAttribute('aria-live', 'polite');
      liveRegion.setAttribute('aria-atomic', 'true');
      liveRegion.className = 'sr-only';
      document.body.appendChild(liveRegion);
      regionRef.current = liveRegion;
      mounted.current = true;
    }

    return () => {
      if (regionRef.current && regionRef.current.parentNode) {
        regionRef.current.parentNode.removeChild(regionRef.current);
      }
    };
  }, []);

  const announce = (message: string, politeness: 'polite' | 'assertive' = 'polite') => {
    if (regionRef.current) {
      regionRef.current.setAttribute('aria-live', politeness);
      regionRef.current.textContent = '';
      setTimeout(() => {
        if (regionRef.current) {
          regionRef.current.textContent = message;
        }
      }, 100);
    }
  };

  return { announce };
}