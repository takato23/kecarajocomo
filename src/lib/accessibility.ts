/**
 * Accessibility utility functions
 */

// Generate unique IDs for ARIA relationships
export function generateId(prefix: string = 'id'): string {
  return `${prefix}-${Math.random().toString(36).substr(2, 9)}`;
}

// Check if user prefers reduced motion
export function prefersReducedMotion(): boolean {
  if (typeof window === 'undefined') return false;
  const mediaQuery = window.matchMedia('(prefers-reduced-motion: reduce)');
  return mediaQuery.matches;
}

// Check if user prefers high contrast
export function prefersHighContrast(): boolean {
  if (typeof window === 'undefined') return false;
  const mediaQuery = window.matchMedia('(prefers-contrast: high)');
  return mediaQuery.matches;
}

// Get user's preferred color scheme
export function getPreferredColorScheme(): 'light' | 'dark' | 'no-preference' {
  if (typeof window === 'undefined') return 'no-preference';
  
  if (window.matchMedia('(prefers-color-scheme: dark)').matches) {
    return 'dark';
  }
  if (window.matchMedia('(prefers-color-scheme: light)').matches) {
    return 'light';
  }
  return 'no-preference';
}

// Trap focus within an element
export function trapFocus(element: HTMLElement) {
  const focusableElements = element.querySelectorAll(
    'a[href], area[href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), button:not([disabled]), iframe, object, embed, [tabindex="0"], [contenteditable]'
  );
  
  const firstFocusableElement = focusableElements[0] as HTMLElement;
  const lastFocusableElement = focusableElements[focusableElements.length - 1] as HTMLElement;

  const handleTabKey = (e: KeyboardEvent) => {
    if (e.key !== 'Tab') return;

    if (e.shiftKey) {
      if (document.activeElement === firstFocusableElement) {
        lastFocusableElement.focus();
        e.preventDefault();
      }
    } else {
      if (document.activeElement === lastFocusableElement) {
        firstFocusableElement.focus();
        e.preventDefault();
      }
    }
  };

  element.addEventListener('keydown', handleTabKey);
  firstFocusableElement?.focus();

  return () => {
    element.removeEventListener('keydown', handleTabKey);
  };
}

// Announce message to screen readers
export function announce(message: string, priority: 'polite' | 'assertive' = 'polite') {
  const announcement = document.createElement('div');
  announcement.setAttribute('aria-live', priority);
  announcement.setAttribute('aria-atomic', 'true');
  announcement.setAttribute('class', 'sr-only');
  announcement.textContent = message;

  document.body.appendChild(announcement);

  setTimeout(() => {
    document.body.removeChild(announcement);
  }, 1000);
}

// Alias for announce function (for backward compatibility)
export const announceToScreenReader = announce;

// Format time for screen readers
export function formatTimeForScreenReader(minutes: number): string {
  const hours = Math.floor(minutes / 60);
  const mins = minutes % 60;
  
  if (hours > 0) {
    return `${hours} hour${hours > 1 ? 's' : ''} and ${mins} minute${mins !== 1 ? 's' : ''}`;
  }
  return `${mins} minute${mins !== 1 ? 's' : ''}`;
}

// Create ARIA description from multiple sources
export function createAriaDescription(...descriptions: (string | undefined)[]): string | undefined {
  const filtered = descriptions.filter(Boolean);
  return filtered.length > 0 ? filtered.join('. ') : undefined;
}

// Check if element is visible
export function isElementVisible(element: HTMLElement): boolean {
  const rect = element.getBoundingClientRect();
  const style = window.getComputedStyle(element);
  
  return (
    rect.width > 0 &&
    rect.height > 0 &&
    style.display !== 'none' &&
    style.visibility !== 'hidden' &&
    style.opacity !== '0'
  );
}

// Get focusable elements within a container
export function getFocusableElements(container: HTMLElement): HTMLElement[] {
  const elements = container.querySelectorAll(
    'a[href], area[href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), button:not([disabled]), iframe, object, embed, [tabindex]:not([tabindex="-1"]), [contenteditable]'
  );
  
  return Array.from(elements).filter(el => isElementVisible(el as HTMLElement)) as HTMLElement[];
}

// Get first focusable element
export function getFirstFocusableElement(container: HTMLElement): HTMLElement | null {
  const elements = getFocusableElements(container);
  return elements.length > 0 ? elements[0] : null;
}

// Get last focusable element
export function getLastFocusableElement(container: HTMLElement): HTMLElement | null {
  const elements = getFocusableElements(container);
  return elements.length > 0 ? elements[elements.length - 1] : null;
}

// Alias for getFocusableElements (for backward compatibility)
export const getAllFocusableElements = getFocusableElements;