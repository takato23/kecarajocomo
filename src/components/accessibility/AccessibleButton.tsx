'use client';

import { forwardRef, ButtonHTMLAttributes } from 'react';

import { cn } from '@/lib/utils';

import { ScreenReaderOnly } from './ScreenReaderOnly';

interface AccessibleButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  variant?: 'primary' | 'secondary' | 'outline' | 'ghost' | 'danger';
  size?: 'sm' | 'md' | 'lg';
  loading?: boolean;
  loadingText?: string;
  icon?: React.ReactNode;
  iconPosition?: 'left' | 'right';
  fullWidth?: boolean;
  ariaLabel?: string;
  ariaPressed?: boolean;
  ariaExpanded?: boolean;
  ariaControls?: string;
  srOnlyText?: string;
}

const buttonVariants = {
  primary: 'bg-primary text-primary-foreground hover:bg-primary/90 focus:ring-primary',
  secondary: 'bg-secondary text-secondary-foreground hover:bg-secondary/80 focus:ring-secondary',
  outline: 'border border-input bg-background hover:bg-accent hover:text-accent-foreground focus:ring-accent',
  ghost: 'hover:bg-accent hover:text-accent-foreground focus:ring-accent',
  danger: 'bg-destructive text-destructive-foreground hover:bg-destructive/90 focus:ring-destructive',
};

const buttonSizes = {
  sm: 'h-8 px-3 text-sm',
  md: 'h-10 px-4',
  lg: 'h-12 px-6 text-lg',
};

export const AccessibleButton = forwardRef<HTMLButtonElement, AccessibleButtonProps>(
  (
    {
      className,
      variant = 'primary',
      size = 'md',
      loading = false,
      loadingText = 'Loading...',
      icon,
      iconPosition = 'left',
      fullWidth = false,
      ariaLabel,
      ariaPressed,
      ariaExpanded,
      ariaControls,
      srOnlyText,
      disabled,
      children,
      ...props
    },
    ref
  ) => {
    const isDisabled = disabled || loading;

    return (
      <button
        ref={ref}
        className={cn(
          'inline-flex items-center justify-center rounded-md font-medium transition-colors',
          'focus:outline-none focus:ring-2 focus:ring-offset-2',
          'disabled:pointer-events-none disabled:opacity-50',
          buttonVariants[variant],
          buttonSizes[size],
          fullWidth && 'w-full',
          className
        )}
        disabled={isDisabled}
        aria-label={ariaLabel}
        aria-pressed={ariaPressed}
        aria-expanded={ariaExpanded}
        aria-controls={ariaControls}
        aria-busy={loading}
        {...props}
      >
        {loading ? (
          <>
            <span className="mr-2 h-4 w-4 animate-spin rounded-full border-2 border-current border-t-transparent" />
            <span aria-live="polite">{loadingText}</span>
          </>
        ) : (
          <>
            {icon && iconPosition === 'left' && (
              <span className="mr-2" aria-hidden="true">
                {icon}
              </span>
            )}
            {children}
            {icon && iconPosition === 'right' && (
              <span className="ml-2" aria-hidden="true">
                {icon}
              </span>
            )}
          </>
        )}
        {srOnlyText && <ScreenReaderOnly>{srOnlyText}</ScreenReaderOnly>}
      </button>
    );
  }
);

AccessibleButton.displayName = 'AccessibleButton';