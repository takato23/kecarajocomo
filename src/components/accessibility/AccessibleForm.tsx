'use client';

import { forwardRef, InputHTMLAttributes, TextareaHTMLAttributes, SelectHTMLAttributes } from 'react';

import { cn } from '@/lib/utils';
import { generateId } from '@/lib/accessibility';


// Accessible Input Component
interface AccessibleInputProps extends InputHTMLAttributes<HTMLInputElement> {
  label: string;
  error?: string;
  hint?: string;
  srOnly?: boolean;
}

export const AccessibleInput = forwardRef<HTMLInputElement, AccessibleInputProps>(
  ({ label, error, hint, srOnly = false, id, className, required, ...props }, ref) => {
    const inputId = id || generateId('input');
    const errorId = error ? generateId('error') : undefined;
    const hintId = hint ? generateId('hint') : undefined;

    return (
      <div className="space-y-2">
        <label 
          htmlFor={inputId} 
          className={cn(
            "text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70",
            srOnly && "sr-only"
          )}
        >
          {label}
          {required && <span className="text-destructive ml-1" aria-label="required">*</span>}
        </label>
        
        {hint && (
          <p id={hintId} className="text-sm text-muted-foreground">
            {hint}
          </p>
        )}
        
        <input
          ref={ref}
          id={inputId}
          className={cn(
            "flex h-10 w-full rounded-md border border-input bg-background px-3 py-2",
            "text-sm ring-offset-background file:border-0 file:bg-transparent",
            "file:text-sm file:font-medium placeholder:text-muted-foreground",
            "focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring",
            "focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50",
            error && "border-destructive focus-visible:ring-destructive",
            className
          )}
          aria-invalid={!!error}
          aria-describedby={cn(errorId, hintId)}
          aria-required={required}
          {...props}
        />
        
        {error && (
          <p id={errorId} className="text-sm text-destructive" role="alert">
            {error}
          </p>
        )}
      </div>
    );
  }
);

AccessibleInput.displayName = 'AccessibleInput';

// Accessible Textarea Component
interface AccessibleTextareaProps extends TextareaHTMLAttributes<HTMLTextAreaElement> {
  label: string;
  error?: string;
  hint?: string;
  srOnly?: boolean;
}

export const AccessibleTextarea = forwardRef<HTMLTextAreaElement, AccessibleTextareaProps>(
  ({ label, error, hint, srOnly = false, id, className, required, ...props }, ref) => {
    const textareaId = id || generateId('textarea');
    const errorId = error ? generateId('error') : undefined;
    const hintId = hint ? generateId('hint') : undefined;

    return (
      <div className="space-y-2">
        <label 
          htmlFor={textareaId} 
          className={cn(
            "text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70",
            srOnly && "sr-only"
          )}
        >
          {label}
          {required && <span className="text-destructive ml-1" aria-label="required">*</span>}
        </label>
        
        {hint && (
          <p id={hintId} className="text-sm text-muted-foreground">
            {hint}
          </p>
        )}
        
        <textarea
          ref={ref}
          id={textareaId}
          className={cn(
            "flex min-h-[80px] w-full rounded-md border border-input bg-background px-3 py-2",
            "text-sm ring-offset-background placeholder:text-muted-foreground",
            "focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring",
            "focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50",
            error && "border-destructive focus-visible:ring-destructive",
            className
          )}
          aria-invalid={!!error}
          aria-describedby={cn(errorId, hintId)}
          aria-required={required}
          {...props}
        />
        
        {error && (
          <p id={errorId} className="text-sm text-destructive" role="alert">
            {error}
          </p>
        )}
      </div>
    );
  }
);

AccessibleTextarea.displayName = 'AccessibleTextarea';

// Accessible Select Component
interface AccessibleSelectProps extends SelectHTMLAttributes<HTMLSelectElement> {
  label: string;
  error?: string;
  hint?: string;
  srOnly?: boolean;
  options: Array<{ value: string; label: string; disabled?: boolean }>;
}

export const AccessibleSelect = forwardRef<HTMLSelectElement, AccessibleSelectProps>(
  ({ label, error, hint, srOnly = false, id, className, required, options, ...props }, ref) => {
    const selectId = id || generateId('select');
    const errorId = error ? generateId('error') : undefined;
    const hintId = hint ? generateId('hint') : undefined;

    return (
      <div className="space-y-2">
        <label 
          htmlFor={selectId} 
          className={cn(
            "text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70",
            srOnly && "sr-only"
          )}
        >
          {label}
          {required && <span className="text-destructive ml-1" aria-label="required">*</span>}
        </label>
        
        {hint && (
          <p id={hintId} className="text-sm text-muted-foreground">
            {hint}
          </p>
        )}
        
        <select
          ref={ref}
          id={selectId}
          className={cn(
            "flex h-10 w-full rounded-md border border-input bg-background px-3 py-2",
            "text-sm ring-offset-background",
            "focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring",
            "focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50",
            error && "border-destructive focus-visible:ring-destructive",
            className
          )}
          aria-invalid={!!error}
          aria-describedby={cn(errorId, hintId)}
          aria-required={required}
          {...props}
        >
          {options.map((option) => (
            <option 
              key={option.value} 
              value={option.value} 
              disabled={option.disabled}
            >
              {option.label}
            </option>
          ))}
        </select>
        
        {error && (
          <p id={errorId} className="text-sm text-destructive" role="alert">
            {error}
          </p>
        )}
      </div>
    );
  }
);

AccessibleSelect.displayName = 'AccessibleSelect';

// Accessible Checkbox Component
interface AccessibleCheckboxProps extends InputHTMLAttributes<HTMLInputElement> {
  label: string;
  error?: string;
  hint?: string;
}

export const AccessibleCheckbox = forwardRef<HTMLInputElement, AccessibleCheckboxProps>(
  ({ label, error, hint, id, className, ...props }, ref) => {
    const checkboxId = id || generateId('checkbox');
    const errorId = error ? generateId('error') : undefined;
    const hintId = hint ? generateId('hint') : undefined;

    return (
      <div className="space-y-2">
        <div className="flex items-center space-x-2">
          <input
            ref={ref}
            type="checkbox"
            id={checkboxId}
            className={cn(
              "h-4 w-4 rounded border-gray-300 text-primary",
              "focus:ring-2 focus:ring-primary focus:ring-offset-2",
              "disabled:cursor-not-allowed disabled:opacity-50",
              className
            )}
            aria-invalid={!!error}
            aria-describedby={cn(errorId, hintId)}
            {...props}
          />
          <label 
            htmlFor={checkboxId} 
            className="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
          >
            {label}
          </label>
        </div>
        
        {hint && (
          <p id={hintId} className="text-sm text-muted-foreground ml-6">
            {hint}
          </p>
        )}
        
        {error && (
          <p id={errorId} className="text-sm text-destructive ml-6" role="alert">
            {error}
          </p>
        )}
      </div>
    );
  }
);

AccessibleCheckbox.displayName = 'AccessibleCheckbox';