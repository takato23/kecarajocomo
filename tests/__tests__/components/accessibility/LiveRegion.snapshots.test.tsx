import React from 'react';
import { render } from '@testing-library/react';
import { LiveRegion } from '@/components/accessibility/LiveRegion';

// Mock setTimeout for consistent snapshots
global.setTimeout = jest.fn((fn: () => void) => {
  fn();
}) as any;

describe('LiveRegion Snapshots', () => {
  describe('Basic LiveRegion', () => {
    it('should render basic live region with message', () => {
      const { container } = render(
        <LiveRegion message="Hello screen readers" />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render empty live region with no message', () => {
      const { container } = render(
        <LiveRegion message="" />
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('LiveRegion Politeness Levels', () => {
    it('should render polite live region', () => {
      const { container } = render(
        <LiveRegion message="Polite announcement" politeness="polite" />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render assertive live region', () => {
      const { container } = render(
        <LiveRegion message="Assertive announcement" politeness="assertive" />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render live region with default politeness', () => {
      const { container } = render(
        <LiveRegion message="Default politeness announcement" />
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('LiveRegion Clear Options', () => {
    it('should render live region with clearOnUnmount enabled', () => {
      const { container } = render(
        <LiveRegion 
          message="Clear on unmount enabled" 
          clearOnUnmount={true} 
        />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render live region with clearOnUnmount disabled', () => {
      const { container } = render(
        <LiveRegion 
          message="Clear on unmount disabled" 
          clearOnUnmount={false} 
        />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render live region with default clearOnUnmount', () => {
      const { container } = render(
        <LiveRegion message="Default clear on unmount" />
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('LiveRegion with Different Message Types', () => {
    it('should render live region with success message', () => {
      const { container } = render(
        <LiveRegion 
          message="Success: Your data has been saved successfully" 
          politeness="polite" 
        />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render live region with error message', () => {
      const { container } = render(
        <LiveRegion 
          message="Error: Unable to save your data. Please try again." 
          politeness="assertive" 
        />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render live region with warning message', () => {
      const { container } = render(
        <LiveRegion 
          message="Warning: This action cannot be undone" 
          politeness="assertive" 
        />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render live region with info message', () => {
      const { container } = render(
        <LiveRegion 
          message="Info: 5 items remaining in your cart" 
          politeness="polite" 
        />
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('LiveRegion with Loading States', () => {
    it('should render live region with loading message', () => {
      const { container } = render(
        <LiveRegion 
          message="Loading your data, please wait..." 
          politeness="polite" 
        />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render live region with progress message', () => {
      const { container } = render(
        <LiveRegion 
          message="Progress: 75% complete" 
          politeness="polite" 
        />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render live region with completion message', () => {
      const { container } = render(
        <LiveRegion 
          message="Task completed successfully" 
          politeness="polite" 
        />
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('LiveRegion with Form-related Messages', () => {
    it('should render live region with form validation message', () => {
      const { container } = render(
        <LiveRegion 
          message="Please enter a valid email address" 
          politeness="assertive" 
        />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render live region with form submission message', () => {
      const { container } = render(
        <LiveRegion 
          message="Form submitted successfully" 
          politeness="polite" 
        />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render live region with field help message', () => {
      const { container } = render(
        <LiveRegion 
          message="Password must be at least 8 characters long" 
          politeness="polite" 
        />
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('LiveRegion with Navigation Messages', () => {
    it('should render live region with page navigation message', () => {
      const { container } = render(
        <LiveRegion 
          message="Navigated to Dashboard page" 
          politeness="polite" 
        />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render live region with content update message', () => {
      const { container } = render(
        <LiveRegion 
          message="Content updated: 3 new items added" 
          politeness="polite" 
        />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render live region with search results message', () => {
      const { container } = render(
        <LiveRegion 
          message="Search completed: 12 results found" 
          politeness="polite" 
        />
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('LiveRegion with Complex Messages', () => {
    it('should render live region with multi-line message', () => {
      const { container } = render(
        <LiveRegion 
          message="Recipe saved successfully. You can find it in your recipe collection. Share it with friends or make it again later." 
          politeness="polite" 
        />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render live region with numbered list message', () => {
      const { container } = render(
        <LiveRegion 
          message="3 errors found: 1. Name is required. 2. Email is invalid. 3. Password is too short." 
          politeness="assertive" 
        />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render live region with dynamic content message', () => {
      const { container } = render(
        <LiveRegion 
          message="Cart updated: 2 items, total $45.99" 
          politeness="polite" 
        />
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('LiveRegion with Special Characters', () => {
    it('should render live region with special characters', () => {
      const { container } = render(
        <LiveRegion 
          message="Price: $29.99 (was $39.99) - 25% off!" 
          politeness="polite" 
        />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render live region with unicode characters', () => {
      const { container } = render(
        <LiveRegion 
          message="Recipe saved: Crème brûlée with café notes ✨" 
          politeness="polite" 
        />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render live region with HTML entities', () => {
      const { container } = render(
        <LiveRegion 
          message="Data imported: 5 recipes & 12 ingredients" 
          politeness="polite" 
        />
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('LiveRegion Edge Cases', () => {
    it('should render live region with very long message', () => {
      const { container } = render(
        <LiveRegion 
          message="This is a very long message that contains a lot of information and might be used to test how the live region handles lengthy announcements that could potentially be truncated or cause performance issues in screen readers but should still be announced properly to users who rely on assistive technologies."
          politeness="polite" 
        />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render live region with whitespace message', () => {
      const { container } = render(
        <LiveRegion 
          message="   Message with whitespace   " 
          politeness="polite" 
        />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render live region with single character message', () => {
      const { container } = render(
        <LiveRegion 
          message="✓" 
          politeness="polite" 
        />
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});