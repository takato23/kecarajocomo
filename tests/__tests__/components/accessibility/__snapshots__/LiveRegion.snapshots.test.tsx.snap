// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LiveRegion Snapshots Basic LiveRegion should render basic live region with message 1`] = `
<div
  aria-atomic="true"
  aria-live="polite"
  class="sr-only"
  role="status"
>
  Hello screen readers
</div>
`;

exports[`LiveRegion Snapshots Basic LiveRegion should render empty live region with no message 1`] = `
<div
  aria-atomic="true"
  aria-live="polite"
  class="sr-only"
  role="status"
/>
`;

exports[`LiveRegion Snapshots LiveRegion Clear Options should render live region with clearOnUnmount disabled 1`] = `
<div
  aria-atomic="true"
  aria-live="polite"
  class="sr-only"
  role="status"
>
  Clear on unmount disabled
</div>
`;

exports[`LiveRegion Snapshots LiveRegion Clear Options should render live region with clearOnUnmount enabled 1`] = `
<div
  aria-atomic="true"
  aria-live="polite"
  class="sr-only"
  role="status"
>
  Clear on unmount enabled
</div>
`;

exports[`LiveRegion Snapshots LiveRegion Clear Options should render live region with default clearOnUnmount 1`] = `
<div
  aria-atomic="true"
  aria-live="polite"
  class="sr-only"
  role="status"
>
  Default clear on unmount
</div>
`;

exports[`LiveRegion Snapshots LiveRegion Edge Cases should render live region with single character message 1`] = `
<div
  aria-atomic="true"
  aria-live="polite"
  class="sr-only"
  role="status"
>
  ✓
</div>
`;

exports[`LiveRegion Snapshots LiveRegion Edge Cases should render live region with very long message 1`] = `
<div
  aria-atomic="true"
  aria-live="polite"
  class="sr-only"
  role="status"
>
  This is a very long message that contains a lot of information and might be used to test how the live region handles lengthy announcements that could potentially be truncated or cause performance issues in screen readers but should still be announced properly to users who rely on assistive technologies.
</div>
`;

exports[`LiveRegion Snapshots LiveRegion Edge Cases should render live region with whitespace message 1`] = `
<div
  aria-atomic="true"
  aria-live="polite"
  class="sr-only"
  role="status"
>
     Message with whitespace   
</div>
`;

exports[`LiveRegion Snapshots LiveRegion Politeness Levels should render assertive live region 1`] = `
<div
  aria-atomic="true"
  aria-live="assertive"
  class="sr-only"
  role="status"
>
  Assertive announcement
</div>
`;

exports[`LiveRegion Snapshots LiveRegion Politeness Levels should render live region with default politeness 1`] = `
<div
  aria-atomic="true"
  aria-live="polite"
  class="sr-only"
  role="status"
>
  Default politeness announcement
</div>
`;

exports[`LiveRegion Snapshots LiveRegion Politeness Levels should render polite live region 1`] = `
<div
  aria-atomic="true"
  aria-live="polite"
  class="sr-only"
  role="status"
>
  Polite announcement
</div>
`;

exports[`LiveRegion Snapshots LiveRegion with Complex Messages should render live region with dynamic content message 1`] = `
<div
  aria-atomic="true"
  aria-live="polite"
  class="sr-only"
  role="status"
>
  Cart updated: 2 items, total $45.99
</div>
`;

exports[`LiveRegion Snapshots LiveRegion with Complex Messages should render live region with multi-line message 1`] = `
<div
  aria-atomic="true"
  aria-live="polite"
  class="sr-only"
  role="status"
>
  Recipe saved successfully. You can find it in your recipe collection. Share it with friends or make it again later.
</div>
`;

exports[`LiveRegion Snapshots LiveRegion with Complex Messages should render live region with numbered list message 1`] = `
<div
  aria-atomic="true"
  aria-live="assertive"
  class="sr-only"
  role="status"
>
  3 errors found: 1. Name is required. 2. Email is invalid. 3. Password is too short.
</div>
`;

exports[`LiveRegion Snapshots LiveRegion with Different Message Types should render live region with error message 1`] = `
<div
  aria-atomic="true"
  aria-live="assertive"
  class="sr-only"
  role="status"
>
  Error: Unable to save your data. Please try again.
</div>
`;

exports[`LiveRegion Snapshots LiveRegion with Different Message Types should render live region with info message 1`] = `
<div
  aria-atomic="true"
  aria-live="polite"
  class="sr-only"
  role="status"
>
  Info: 5 items remaining in your cart
</div>
`;

exports[`LiveRegion Snapshots LiveRegion with Different Message Types should render live region with success message 1`] = `
<div
  aria-atomic="true"
  aria-live="polite"
  class="sr-only"
  role="status"
>
  Success: Your data has been saved successfully
</div>
`;

exports[`LiveRegion Snapshots LiveRegion with Different Message Types should render live region with warning message 1`] = `
<div
  aria-atomic="true"
  aria-live="assertive"
  class="sr-only"
  role="status"
>
  Warning: This action cannot be undone
</div>
`;

exports[`LiveRegion Snapshots LiveRegion with Form-related Messages should render live region with field help message 1`] = `
<div
  aria-atomic="true"
  aria-live="polite"
  class="sr-only"
  role="status"
>
  Password must be at least 8 characters long
</div>
`;

exports[`LiveRegion Snapshots LiveRegion with Form-related Messages should render live region with form submission message 1`] = `
<div
  aria-atomic="true"
  aria-live="polite"
  class="sr-only"
  role="status"
>
  Form submitted successfully
</div>
`;

exports[`LiveRegion Snapshots LiveRegion with Form-related Messages should render live region with form validation message 1`] = `
<div
  aria-atomic="true"
  aria-live="assertive"
  class="sr-only"
  role="status"
>
  Please enter a valid email address
</div>
`;

exports[`LiveRegion Snapshots LiveRegion with Loading States should render live region with completion message 1`] = `
<div
  aria-atomic="true"
  aria-live="polite"
  class="sr-only"
  role="status"
>
  Task completed successfully
</div>
`;

exports[`LiveRegion Snapshots LiveRegion with Loading States should render live region with loading message 1`] = `
<div
  aria-atomic="true"
  aria-live="polite"
  class="sr-only"
  role="status"
>
  Loading your data, please wait...
</div>
`;

exports[`LiveRegion Snapshots LiveRegion with Loading States should render live region with progress message 1`] = `
<div
  aria-atomic="true"
  aria-live="polite"
  class="sr-only"
  role="status"
>
  Progress: 75% complete
</div>
`;

exports[`LiveRegion Snapshots LiveRegion with Navigation Messages should render live region with content update message 1`] = `
<div
  aria-atomic="true"
  aria-live="polite"
  class="sr-only"
  role="status"
>
  Content updated: 3 new items added
</div>
`;

exports[`LiveRegion Snapshots LiveRegion with Navigation Messages should render live region with page navigation message 1`] = `
<div
  aria-atomic="true"
  aria-live="polite"
  class="sr-only"
  role="status"
>
  Navigated to Dashboard page
</div>
`;

exports[`LiveRegion Snapshots LiveRegion with Navigation Messages should render live region with search results message 1`] = `
<div
  aria-atomic="true"
  aria-live="polite"
  class="sr-only"
  role="status"
>
  Search completed: 12 results found
</div>
`;

exports[`LiveRegion Snapshots LiveRegion with Special Characters should render live region with HTML entities 1`] = `
<div
  aria-atomic="true"
  aria-live="polite"
  class="sr-only"
  role="status"
>
  Data imported: 5 recipes & 12 ingredients
</div>
`;

exports[`LiveRegion Snapshots LiveRegion with Special Characters should render live region with special characters 1`] = `
<div
  aria-atomic="true"
  aria-live="polite"
  class="sr-only"
  role="status"
>
  Price: $29.99 (was $39.99) - 25% off!
</div>
`;

exports[`LiveRegion Snapshots LiveRegion with Special Characters should render live region with unicode characters 1`] = `
<div
  aria-atomic="true"
  aria-live="polite"
  class="sr-only"
  role="status"
>
  Recipe saved: Crème brûlée with café notes ✨
</div>
`;
